pkg/executor/cluster.go

```diff
@@ -227,3 +230,44 @@ func compactMembers(
 	}
 	*m = (*m)[:numMembers-numRemovedMembers]
 }
+
+// UpdateStorageInitProgress parses rsync output
+// and sets the current % progress to memberStatus.StateDetail.StorageInitPercent field
+func UpdateStorageInitProgress(
+	reqLogger logr.Logger,
+	cr *kdv1.KubeDirectorCluster,
+	memberStatus *kdv1.MemberStatus,
+	initContainerStatus corev1.ContainerStatus,
+) {
+
+	var rsyncStatusStrB strings.Builder
+	progressBarFile := fmt.Sprintf("/mnt%s", kubedirectorInitProgressBar)
+
+	read, err := ReadFile(
+		reqLogger,
+		cr,
+		cr.Namespace,
+		(*memberStatus).Pod,
+		initContainerStatus.ContainerID,
+		initContainerName,
+		progressBarFile,
+		&rsyncStatusStrB,
+	)
+	if err != nil {
+		shared.LogErrorf(
+			reqLogger,
+			err,
+			cr,
+			shared.EventReasonCluster,
+			"failed to read %s",
+			progressBarFile,
+		)
+	}
+
+	if read {
+		lines := strings.Split(rsyncStatusStrB.String(), "\r")
+		lastLine := lines[len(lines)-1]
+		fields := strings.Fields(lastLine)
```

Comment:
Overall this looks good, the only thing I'm really wondering about is the robustness of this rsync output parsing.

I'm wondering if maybe we should just stick the whole progress line into this property... not only would that be a little less brittle (although probably rsync does have a vested interest in keeping that progress output in a fixed format), it would also give a bit more information about things like amount copied so far, rate of copy, time spent so far, etc.

What do you think?

Reply:
Do we really need this info out of logs? Sure, we may print the whole `lastLine value`

Reply:
Yeah, I think just that whole last line. Specifically this sort of thing:
```
105.45M  13%  602.83kB/s    0:02:50 (xfr#495, ir-chk=1020/3825)
```
Showing total amount copied, and especially rate of copy and time spent so far, are at least as valuable as the current percent copied. The last bit of parenthetical stuff isn't necessarily as valuable but I don't know that we need to worry about chopping it out. I would be OK with chopping off the parenthetical portion but also OK with leaving it there.

As a bonus, if we don't bother parsing this we will be resilient to any changes in rsync progress output format.

Reply:
Done